docs/grid-functionalities/Row-based-edit.md

```diff
@@ -0,0 +1,70 @@
+- [Disable External Button when having Empty Selection](#disable-external-button-when-having-empty-selection)
+- [Change Row Selections](#change-row-selections)
+- Troubleshooting
+  - [Adding a Column dynamically is removing the Row Selection, why is that?](#adding-a-column-dynamically-is-removing-the-row-selection-why-is-that)
+
+### Description
+The Row based editing plugin makes it possible to keep the grid readonly except for rows which the user explicitely toggles into editmode.
+
+**Note:** This plugin enforces the use of the `autoEdit` option and will turn it on with a console warning if its not already.
+
+### Demo
+[Demo Page](https://ghiscoding.github.io/slickgrid-universal/#/example22) / [Demo ViewModel](https://github.com/ghiscoding/slickgrid-universal/blob/master/examples/webpack-demo-vanilla-bundle/src/examples/example22.ts)
+
+## The action column
+A new column is rendered that shows an edit/delete button. If the user clicks on edit, a save and cancel button are shown instead and the row toggles into editmode. By default as the last column but you can override it with the option `columnIndexPosition`. Additionally it's default column id can be overriden using the opiton `columnId`. Furthermore, you can also override the columns label via the `actionsColumnLabel` property.
+
+### Single or multiple editable rows
+By default you can only toggle a single row into editmode. If you set the option `allowMultipleRows` to `true` on the other hand, you can toggle as many as you want.
+
+### Configuring the action buttons
+You can override the styling, the hover text as well as whether a prompt — and with what text — should be shown. It is done by overriding the `actionButtons` property of the [plugins options](https://github.com/ghiscoding/slickgrid-universal/blob/master/packages/common/src/interfaces/rowBasedEditOption.interface.ts).
+
+## Support for the Excel Copy Buffer Plugin
+If the [Excel Copy Buffer Plugin](grid-functionalities/excel-copy-buffer.md) is configured, the Row based edting pluging will override it's behavior by denying pastes on all cells not within a editmode row. Nevertheless, any existing `BeforePasteCellHandler` will be respected.
+
+## How the plugin works
+The idea of the plugin is to focus the users editing experience on specific individual rows and and save them individually. This is achieved by letting the user toggle one or more rows into editmode. Now changes can be made to those rows and will be highlighted and tracked. The user may cancel the editmode at any time and revert all cells changes. If the save button is pressed on the other hand an `onBeforeRowUpdated` hook, which you define via plugin options, is called and expects a `Promise<boolean>`. In that method you'd typically write the changes to your backend and return either true or false based on the operations outcome. If a negative boolean is returned the editmode is kept, otherwise the row applies the changes and toggles back into readonly mode.
```

Comment:
"...and toggles back into readonly mode."  so just to confirm, when this edit mode is used, the grid itself cannot be edited the typical way (click on any cell), we really need to use the edit mode, is that it? We might need to add a sentence explaining what I just asked, it wasn't clear to me

Reply:
exactly. i've added that in the description at the begin. do you want to repeat the info here to be on the safe side?

Reply:
yeah might as well to be clearer since I didn't notice the top description